### DISCUSSION TITLE
Handling invoice.payment_succeeded event with Stripe checkout Laravel Cashier

### DISCUSSION THREAD

**Discussion by sts-ryan-holton** [laravel/framework#51232] [Q&A]
I'm migrating my Laravel Cashier setup over to utilise the Stripe checkout feature. My platform offers both: subscriptions and one-off charges. One of these one-off charges is the ability to replenish SMS credits in the user's account, originally, I was using the `invoicePrice` method and pass through some metadata:

```php
$addonItemPrice = Cashier::stripe()->prices->retrieve($addonItem->stripe_id);
Auth::user()->invoicePrice(
    $addonItem->stripe_id, 1, [
        'metadata' => [
            'is_addon' => true,
            'is_sms_addon' => true,
            'created_at' => Carbon::now(),
            'user_id' => Auth::id(),
            'addon_id' => $addonItem->id,
            'addon_quantity' => $addonItem->quantity,
            'apply_credit_balance' => true,
        ],
    ]
);
```

I'm then listening for a webhook event for `invoice.payment_succeeded` to know for sure to add SMS credits to their account:

```php
/**
 * Handle received Stripe webhooks.
 */
public function handle(WebhookReceived $event)
{
    match ($event->payload['type']) {
        'customer.subscription.created' => $this->handleSubscriptionCreated($event->payload),
        'invoice.payment_succeeded' => $this->handleInvoicePaymentSucceeded($event->payload),
        default => null
    };
}
```

However, I'm now migrating to using Stripe's checkout, so it now looks like:

```php
$charge = $user->checkout([$addonItemPrice->id => 1], [
    'success_url' => "$frontendURL"."account/billing/checkout/success/",
    'cancel_url' => "$frontendURL"."account/billing/add-ons/",
    'metadata' => [
        'is_addon' => true,
        'is_sms_addon' => true,
        'created_at' => Carbon::now(),
        'user_id' => Auth::id(),
        'addon_id' => $addonItem->id,
        'addon_quantity' => $addonItem->quantity,
        'apply_credit_balance' => true,
    ],
]);
```

But I'm not getting an `invoice.payment_succeeded` event at all now. It seems that my metadata is now associated with the `
checkout.session.completed` event.

Should I migrate to this new event, or is this going to be unreliable? I only want such event to trigger when the invoice payment has succeeded

**Comment by OliveLeb** [comment]
I don't think it has anything to do with cashier. I believe that stripe checkout do not generate invoices hence do not generate invoice related events so the checkout one must be the one :)

**Reply by sts-ryan-holton** [reply]
I had to add the following, which worked:

```
'invoice_creation' => [
   'enabled' => true,
   'invoice_data' => [
      'metadata' => [
         // your metadata
      ]
   ]
]
````

**Reply by OliveLeb** [reply]
Yes but be careful, checkout invoice creation is not free
